src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -88,7 +88,8 @@ public static class AttributeMatcher
             {
                 if (!context.AllAttributes.ContainsKey(attribute) ||
                     context.AllAttributes[attribute] == null ||
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute] as string))
+                    (typeof(string).IsAssignableFrom(context.AllAttributes[attribute].GetType()) &&
+                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute] as string)))
```

Comment:
:ok:
so the intent here is to treat _any_ value for non-`string` properties as provided?  I guess that makes sense for `bool?` but I'm wondering if it's the correct behaviour in general.  not sure about this...